great_expectations/data_context/data_context.py

```diff
@@ -1849,169 +1636,134 @@ def profile_datasource(self,
+
+    An expectation suite is a collection of expectations ready to be applied to a batch of data. Since
+    in many projects it is useful to have different expectations evaluate in different contexts--profiling
+    vs. testing; warning vs. error; high vs. low compute; ML model or dashboard--suites provide a namespace
+    option for selecting which expectations a DataContext returns.
+
+    In many simple projects, the datasource or generator name may be omitted and the DataContext will infer
+    the correct name when there is no ambiguity.
+
+    Similarly, if no expectation suite name is provided, the DataContext will assume the name "default".
+    """
+
+    # def __init__(self, config, filepath, data_asset_name_delimiter='/'):
+    def __init__(self, context_root_dir=None, data_asset_name_delimiter='/'):
+
+        # #TODO: Factor this out into a helper function in GE. It doesn't belong inside this method.
+        # # determine the "context root directory" - this is the parent of "great_expectations" dir
+        if context_root_dir is None:
+            if os.path.isdir("../notebooks") and os.path.isfile("../great_expectations.yml"):
+                context_root_dir = "../"
+            elif os.path.isdir("./great_expectations") and \
+                    os.path.isfile("./great_expectations/great_expectations.yml"):
+                context_root_dir = "./great_expectations"
+            elif os.path.isdir("./") and os.path.isfile("./great_expectations.yml"):
+                context_root_dir = "./"
+            else:
+                raise DataContextError(
+                    "Unable to locate context root directory. Please provide a directory name."
+                )
+        context_root_directory = os.path.abspath(context_root_dir)
+        self._context_root_directory = context_root_directory
+
+        project_config = self._load_project_config()
+
+        super(DataContext, self).__init__(
+            project_config,
+            context_root_directory,
+            data_asset_name_delimiter,
+        )
+
+    def _load_project_config(self):
+        """Loads the project configuration file."""
+        try:
+            with open(os.path.join(self.root_directory, "great_expectations.yml"), "r") as data:
+                config = yaml.load(data)
+
+                if config["stores"] == None:
+                    config["stores"] = {}
+
+                return DataContextConfig(**config)
+
+        except IOError:
+            raise ConfigNotFoundError(self.root_directory)
+
+    def _save_project_config(self):
+        """Save the current project to disk."""
+        with open(os.path.join(self.root_directory, "great_expectations.yml"), "w") as data:
+            #FIXME: This method is currently Deactivated
+            print(type(self._project_config))
+            config = self._project_config.as_dict()
+            yaml.dump(config, data)
+            pass
+
+    def add_store(self, store_name, store_config):
```

Comment:
Why?

IMO, `add_store`'s goal is to add a store to the DataContext's `stores` property, which is accomplished without returning anything.

At best, returning the store is convenient for the developer, saving them a `my_store = context.stores[my_store_name]` in the next line. At worst, it could be confusing.

Again, I don't have a strong opinion on this one, I just don't see a strong reason to prefer one over the other. Am I missing something?

Reply:
This continues a thread above, but I must be missing something...why would it be confusing?